src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1208 @@
+        if "roberta" in config.tokenizer:
+            self.tokenizer_config = RobertaConfig(
+                vocab_size=config.vocab_size,
+                hidden_size=config.hidden_size,
+                num_hidden_layers=config.num_hidden_layers,
+                num_attention_heads=config.num_attention_heads,
+                intermediate_size=config.hidden_size * config.mlp_ratio,
+                max_position_embeddings=config.max_text_len,
+                hidden_dropout_prob=config.drop_rate,
+                attention_probs_dropout_prob=config.drop_rate,
+            )
+        else:
+            raise ValueError("Incorrect value of tokenizer. Currently roberta tokenizer is supported")
+
+        if config.cross_modal_transform_shared:
+            self.cross_modal_text_transform = nn.Linear(config.input_text_embed_size, config.hidden_size)
+            self.cross_modal_image_transform = nn.Linear(config.input_image_embed_size, config.hidden_size)
+        else:
+            self.cross_modal_text_transform = nn.ModuleList(
+                [nn.Linear(config.input_text_embed_size, config.hidden_size) for _ in range(config.num_hidden_layers)]
+            )
+            self.cross_modal_image_transform = nn.ModuleList(
+                [nn.Linear(config.input_image_embed_size, config.hidden_size) for _ in range(config.num_hidden_layers)]
+            )
+        self.cross_modal_text_transform.apply(self._init_weights)
+        self.cross_modal_image_transform.apply(self._init_weights)
+
+        self.token_type_embeddings = nn.Embedding(2, config.hidden_size)
+        self.token_type_embeddings.apply(self._init_weights)
+
+        if self.is_clip:
+            self.vit_model = BridgeTowerCLIP(
+                config.vit_embed_dim,
+                config.resolution_before,
+                config.vit_layers,
+                config.vit_width,
+                config.vit_patch_size,
+                config.vit_transformer_width,
+                resolution_after=config.image_size,
+                stop_gradient=config.stop_gradient,
+                vit_layernorm_shared=config.vit_layernorm_shared,
+                vit_remove_last=config.vit_remove_last,
+            )
+        else:
+            raise ValueError(
+                "Incorrect value of visual encoder. Currently ViT-B/32, ViT-B/16 or ViT-L/14 are supported."
+            )
+
+        if "roberta" in config.tokenizer:
+            roberta_config = RobertaConfig.from_pretrained(config.tokenizer)
+            self.text_transformer = RobertaModel(roberta_config)
+        else:
+            raise ValueError("Incorrect value of tokenizer. Currently roberta tokenizer is supported")
+
+        if not config.vit_layernorm_shared and config.vit_layernorm_init_from_vit:
+            for ln in self.vit_model.visual.cross_modal_ln_separate:
+                ln.weight.data = self.vit_model.visual.ln_post.weight.data
+                ln.bias.data = self.vit_model.visual.ln_post.bias.data
```

Comment:
Probably better to wrap this in a class method `_init_ln_from_vit`
Or add it on `_init_weights` function if it is doable

Reply:
Handled! Thanks for the feedback.